### PR TITLE
preventing duplicate events on filter control change

### PR DESCRIPTION
**Pull request by Tri4Ty**

### CODE REVIEW

File: verticalFilterBar/filterControls/TextBoxControl.jsx [hunk 1]
```diff
@@ -22,37 +22,47 @@ export default class TextBoxControl extends Component {
 
   componentDidUpdate() {
     let {currentCriterion} = this.props;
-    if (currentCriterion && currentCriterion.values &&
+    if (currentCriterion && currentCriterion.values && 
+      // Object.keys(currentCriterion.values).length === 0 &&
```
Comment (bradhawbaker) [inline verticalFilterBar/filterControls/TextBoxControl.jsx:26]: Dead code

File: verticalFilterBar/filterControls/TextBoxControl.jsx [hunk 2]
```diff
@@ -22,37 +22,47 @@ export default class TextBoxControl extends Component {
...
+    } else if ((!currentCriterion || !currentCriterion.values) && this.state.currentCriterion.values) {
+      // need to clear any previously set values
+      this.handleTextChange({});
     }
   }
 
   handleTextChange(currentCriterion) {
     const value = currentCriterion.values;
-    const {controlId} = this.props;
+    // const {controlId} = this.props;
```
Comment (bradhawbaker) [inline verticalFilterBar/filterControls/TextBoxControl.jsx:39]: Dead code (46, 52)

**Review by bradhawbaker — APPROVED** [review]

File: verticalFilterBar/filterControls/TextBoxControl.jsx (hunk 1 above)
Comment (Tri4Ty) [reply verticalFilterBar/filterControls/TextBoxControl.jsx:26]: currentCriterion.values can be an empty map (i.e. {}) ... that's why I check for Object.keys(currentCriterion.values).length === 0

File: verticalFilterBar/filterControls/TextBoxControl.jsx (hunk 2 above)
Comment (Tri4Ty) [reply verticalFilterBar/filterControls/TextBoxControl.jsx:39]: thanks, I'll clean that up

**Pull request merged**